repository/src/raw/mod.rs

```diff
@@ -112,6 +112,8 @@ pub trait RawRepository: Send + Sync + 'static {
     ) -> Result<CommitHash, Error>;
 
     /// Creates a semantic commit from the currently checked out branch.
+    ///
+    /// It fails if the `diff` is not `Diff::Reserved` or `Diff::None`.
```

Comment:
Should the contents of files in reserved directory change programmatically based on the reserved_state in diff?
For example, make new file about new member or modify files ...

Reply:
As you can notice, both `Diff::Reserved` and `Diff::General` contain an entire new copy of the RS.  You can just overwrite it.